hazelcast/src/main/java/com/hazelcast/internal/usercodedeployment/impl/ClassData.java

```diff
@@ -19,31 +19,48 @@
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import com.hazelcast.nio.serialization.impl.Versioned;
+import com.hazelcast.version.Version;
 
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.hazelcast.internal.cluster.Versions.V3_9;
 
 /**
- * Bytecode of a class.
+ * Carries byte code of a class along with its inner classes.
+ *
  *
  * It's wrapped inside own object as it allows to add additional metadata and maintain compatibility
  * with Hazelcast Rolling Upgrade.
  */
-public class ClassData implements IdentifiedDataSerializable {
+public class ClassData implements IdentifiedDataSerializable, Versioned {
+
+    private Map<String, byte[]> classDefinitions;
 
-    private byte[] classDefinition;
+    //this field needed for compatibility with 3.8
+    private byte[] requestedClassDefinition;
```

Comment:
When the requester is 3.8 `requestedClassDefinition ` can be inner class. That is why I avoided naming them as `innerClassDefinitons` vs `outer`/`main`.  

After 3.9, I was thinking to remove the `requestedClassDefinition` completely and keep everything in `classDefinitions`, simply because there is no need to separate them. 

In short, I avoided naming as you suggested because it might be confusing when trying to understand relation with 3.8. Note that , I was mostly thinking `outer` instead of `main` . 
If that does not seem confusing, I can continue with your suggestion.